### ISSUE TITLE
pkg messages breaks doctests

### ISSUE THREAD

**Issue by Borda** [allegroai/trains#119]
We have some problems with Trains pushing messages via `sys.stdout` instead commonly used `sys.stderr`. The case is with doctests where the output is compared to the expected value and with just minor package update a message text change it breaks our tests like this example:
```
064         >>> logger.log_metrics({"val_loss": 1.23}, step=0)
Expected nothing
Got:
    TRAINS new version available: upgrade to v0.14.1rc2 is recommended!
```
a simples solution on our side is to freeze version but it would limit Trains users so I would appreciate to change the streaming channel or be able to configure it
https://github.com/PyTorchLightning/pytorch-lightning/pull/1087/checks?check_run_id=532067360

**Comment by Borda** [comment]
@bmartinn could we do something here? it is really important for us... :]

**Comment by bmartinn** [comment]
@Borda will routing the message to sys.stderr solve the issue ?

**Comment by Borda** [comment]
it should, it is a similar case as TQDM :]
pls check this example:
```python
import sys

def anyfunc(bool_std):
    """
    >>> anyfunc(True)
    print with `sys.stdout`
    'abc'
    >>> anyfunc(False)
    'abc'
    """
    if bool_std:
        sys.stdout.write("print with `sys.stdout`\n")
    else:
        sys.stderr.write("print with `sys.stderr`\n")
    return "abc"
```

**Comment by bmartinn** [comment]
@Borda, there should not be any (stdout or stderr) prints in the current (master branch) of PyTorch-Lightning. I think this is a specific issue with this PR since it was before we introduced the "bypass mode"/"offline mode" to the `test_trains.py` .
May I suggest merging the test code from the master branch into the PR so that the CI pass ?

**Comment by Borda** [comment]
ok, Ill try but still I would recommend doing this change... :]

**Comment by bmartinn** [comment]
Thanks @Borda , let me repeat here, just so I'm sure I understand your suggestion.
Are you suggesting to have all the *Trains* messages printed on `stderr` instead of `stdout` ?

**Comment by Borda** [comment]
yes or having the stream as a parameter, eg [pprint](https://docs.python.org/3/library/pprint.html#pprint.pprint) so you can define even a file

**Comment by bmartinn** [comment]
Since *Trains* is an auto-magical solution, there is really very little you have to do in order to get full versioning of your experiment.

Just importing and initializing Trains is enough, anything you push to stderr / stdout will be automatically logged to the server, any TensorBoard graph, or MatplotLib plot will be logged as well. *Trains* will also log a reference to your git repository, commit id, uncommitted changes, and all the python packages you are using, all logged automatically.  
Further more, all the models loaded or saved by PyTorch/Keras/TF will be logged as well.
This is still with no additional code on the user part (other than the first import/initialize).

This is why if *Trains* outputs any general purpose information (i.e. not something I'd like to see on `stderr` , since it is not an error) it will be on `stdout`. Any other warning/error is logged using the standard python logger.

Makes sense?

**Comment by Borda** [comment]
well not very happy about it but fine :]
so could you please remake tests in lightning from `doctest` to standard `unittests` :angel:

**Comment by Borda** [comment]
> It is not, this is a general purpose message (that should not appear in "bypass mode") saying you should upgrade your _Trains_ RC package. The fact it was printed just after the logger call is just a coincidence, they are unrelated :)

I understand the purpose but it breaks doctest as with newer installed package version the text might change... and a week ago there was no message at all...

**Comment by bmartinn** [comment]
@Borda, let me stress that again, **Zero messages** will be printed after your PyTorch-Lightning PR 1172 . Regardless of the *Trains* version, or anything else, this is exactly why we added the "bypass mode" :)

**Comment by bmartinn** [comment]
@Borda just checking in, did the CI tests pass? Should I push a temp fix to the PR branch?

**Comment by Borda** [comment]
to be honest it is quite hard to check if the failing PRs are using latest master with the "by-pass" fix or not but just today I say failing some tests because of anonoucing new release...

**Comment by bmartinn** [comment]
@borda, Could you temporarily block the update domain? (Like set the hosts file on the CI machine)
It would eliminate any "version update"

**Comment by Borda** [comment]
that shall be the "by-pass" right?

**Comment by bmartinn** [comment]
Regardless of the "by pass" fix.
It will block the update query, so it will not "know" there is an update, hence no message print.
That said, there are other prints, like experiment ID, that might change, not sure how that affects the CI tests.

I guess the easiest way is still to merge the "by pass" fix...
Just trying to help with this specific older branch PR.

**Comment by bmartinn** [comment]
Fix PyTorchLightning/pytorch-lightning#1379 merged to master
